server/api/classyear/classyear.controller.js

```diff
@@ -219,6 +223,27 @@ function generateCode(codeLength){
   for(var i=0;i<codeLength;i++){
       var character = (Math.floor(Math.random() * characterOptions.length));
       code = code.concat(characterOptions[character.toString()]);
-  }
+  }  
   return code;
 }
+
+//Generating unique code.
+function uniqueDayCode(code,codeLength,callback){  
+  ClassYear.findOne({"dayCodes.code":code})
+    .exec(function(err, classYear){
+      if (err) return callback("error when getting dayCode",null);
+      if(classYear) {
+        return uniqueDayCode(generateCode(codeLength+1),codeLength+1,callback);
```

Comment:
This is ugly. Would be a lot cleaner if generateCode was just called at the beginning of uniqueDayCode. Then you don't need to define the code length twice.